src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -377,6 +377,16 @@ private MsalAccessTokenCacheItem FindAccessTokenCommon
                             .Where(item => item.HomeAccountId.Equals(requestParams.Account?.HomeAccountId.Identifier, StringComparison.OrdinalIgnoreCase))
                             .ToList();
                 }
+
```

Comment:
Thx for explaining, this makes sense now. I suggest you extract this logic in a private method and add comments similar to your example in the email on why this step is needed.